### DISCUSSION TITLE
Setup Process - Operation not supported for a relative URI

### DISCUSSION THREAD

**Discussion by irishman2020** [AdhocAdam/smletsexchangeconnector#449] [Q&A] [answered]
Setting up for the first time. I followed the guides to setup SMLets with native service manager on a fresh install of Server 2019 core and SCSM 2022. After working through a few issues, I'm getting the successfully retrieved an OAuth token log, but then I get this error:

System.Management.Automation.MethodInvocationException: Exception calling "Bind" with "2" argument(s): "This operation is not supported for a relative URI." ---> System.InvalidOperationException: This operation is not supported for a relative URI.
   at System.Uri.get_AbsoluteUri()
   at Microsoft.Exchange.WebServices.Data.ExchangeCredentials.GetUriWithoutSuffix(Uri url)
   at Microsoft.Exchange.WebServices.Data.ExchangeCredentials.AdjustUrl(Uri url)
   at Microsoft.Exchange.WebServices.Data.ExchangeService.PrepareHttpWebRequest(String methodName)
   at Microsoft.Exchange.WebServices.Data.ServiceRequestBase.BuildEwsHttpWebRequest()
   at Microsoft.Exchange.WebServices.Data.ServiceRequestBase.ValidateAndEmitRequest(IEwsHttpWebRequest& request)
   at Microsoft.Exchange.WebServices.Data.MultiResponseServiceRequest`1.Execute()
   at Microsoft.Exchange.WebServices.Data.ExchangeService.BindToFolder(FolderId folderId, PropertySet propertySet)
   at Microsoft.Exchange.WebServices.Data.ExchangeService.BindToFolder[TFolder](FolderId folderId, PropertySet propertySet)
   at CallSite.Target(Closure , CallSite , Type , Object , Object )
   --- End of inner exception stack trace ---
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)

**Comment by irishman2020** [comment]
Started reading every previous discussion and found this one: https://github.com/AdhocAdam/smletsexchangeconnector/discussions/359
I added the Tls12 setting to the script on line 219
`[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
`
I also ran this script to get TLS1.2 set as default and rebooted: https://github.com/beowulf005/TLS-Config
No change so far, still getting the error.

**Reply by AdhocAdam** [reply]
Hrmmm. This is an interesting one for sure and something I'm going to make several assumptions on, so I'm going to tal...type outloud here.

- You're getting an OAuth token, so it sounds like you're using Exchange Online
- If you're using Exchange online and getting a token, you've correctly configured an Azure App Registration
- If you're using Exchange online, you've manually set the Autodiscover address
- You're getting a Bind error and there are 4 ::Bind calls made throughout the connector. I _want to_ make an assumption on which one it's occurring based on the error message, but I'd need to know where did you get this error message? The SMLets Exchange Connector Log or the Operations Manager event log?

**Reply by irishman2020** [reply]
I apologize for not clarifying. 
All your assumptions are correct.
The error comes from the SMLets eventviewer logs. I was getting this:
![image](https://user-images.githubusercontent.com/11076035/227566559-7d5e2bef-9873-4ada-a3b5-9b885e032f06.png)

**Reply by irishman2020** [reply]
As a side note, after working on getting the TLS settings correct for SMLets, I was able to get the Exchange Connector 4.1 working today and pulling emails in from 365. I would still like to leverage SMLets, but I am able to move forward with getting my new server working (previously we were using Server 2016/ SCSM 2016 using Exchange connector to connect to our Exchange 2016 until we moved to Exchange 2019, which broke the connector and started me down this rabbithole...)

**Reply by AdhocAdam** [reply]
No worries!

Alright so that is the bind that is failing on/about 4937 which is responsible for actually retrieving the Inbox.
```powershell
$inboxFolder = [Microsoft.Exchange.WebServices.Data.Folder]::Bind($exchangeService,$inboxFolderName)
```

Without really diving into this, the only reason I can think this is happening is that the Autodiscover URL in the Settings UI is something like:
```text
outlook.office365.com/EWS/Exchange.asmx
```

Instead of:
```text
https://outlook.office365.com/EWS/Exchange.asmx
```

**Reply by irishman2020** [reply]
Oh wow.... so {null} was not acceptable? *picardfacepalm*

**Reply by AdhocAdam** [reply]
This is also on the Wiki after configuring the Azure App in [Scenario 3](https://github.com/AdhocAdam/smletsexchangeconnector/wiki/Configuration-Examples#scenario-3-exchange-onlineoffice-365-with-oauth).

But if it works that's all that matters. 😄

**Reply by AdhocAdam** [reply]
Just wanted to follow up here, everything sorted out?

**Reply by irishman2020** [reply]
I haven't had a chance to test properly. We got it working with the exchange connector, and we had to make a mad rush to do some catchup work and get the rest of the configuration working. I forgot to mark it answered, and I'll do so now. 

Thank you again!

**Reply by AdhocAdam** [reply]
Most excellent. Glad to help 😃

**Reply by AdhocAdam** [reply]
If this project helps you out, consider giving the repo a star

**Comment by irishman2020** [comment, marked as answer]
Issue resolved, problem with exchange url missing in config.
